### ISSUE TITLE
猪猪们，这是你们想要的东西吗？

### ISSUE THREAD

**Issue by BarCodeReader** [lizhengwei1992/Semantic_Human_Matting#52]
这样的TNet？这样的ANet?
![image](https://user-images.githubusercontent.com/39732241/111504493-ad6c8280-8782-11eb-9408-c1ed94e80686.png)

![image](https://user-images.githubusercontent.com/39732241/111328343-1e3f6c00-86a9-11eb-8a36-28b74751f7c5.png)
别想着白嫖了， 自己动手吧。
这个repo的作者已经给你们平了90%的路了，兄弟们。

**Comment by LoveU3tHousand2** [comment]
大佬 我又来烦你了。  我这边在拿另一个模型做测试的时候**只要把尺寸稍微大一点**（大概700x700）的图片给模型处理就会报Cuda内存溢出， 但是相同的图片放到SHM这个项目中就不会。   是什么原因呢？ 因为模型本身就有尺寸大小限制吗， 我把图片cv2.resize 之后就可以放入模型处理，但是缩放之后图片又会失真。 已经在鼓捣了好几天了。也没办法解决。。     
是因为[用一定尺寸上限的图片训练出来的模型，无法处理超过此上限的图片]  ？  
还是说有办法调试解决的 ？
还请大佬指条明路

**Comment by BarCodeReader** [comment]
1》模型不同确实cuda 占用不同，所以这个最直接就是换大一点的显卡
2》为什么一定要end-to-end，你先resize变小，拿去产生trimap， 释放掉cuda内存，然后把trimap放大，拿去产生alpha matte，不可以吗？
3》【用一定尺寸上限的图片训练出来的模型，无法处理超过此上限的图片】不存在这个说法

**Comment by LoveU3tHousand2** [comment]
谢谢大佬的回复啊，  您有空能看看这个吗 https://github.com/microsoft/Bringing-Old-Photos-Back-to-Life/issues/19

我说的就是这个东西， 它已经提供了模型， 但是用这个模型处理图片的时候，尺寸大了就报错。 Global/test.py中 [generated = model.inference(input, mask)] 这一步会捕获异常CUDA OOM
 然后可以追溯到Global/models/NonLocal_feature_mapping_model.py 中。。。
 网络上能查得到的方法基本都尝试过了， 只要一inference大图片就必然显存溢出。 按您说的思路 ，不end-to-end 能有什么办法改一下吗。   我目前能想到的方法就只有把mask和input_image 都resize后放入model.inference(),

**Comment by BarCodeReader** [comment]
这两个模型都不是做的同一件事情，就没得比较了。
用encoder decoder做restoration耗gpu内存是很正常的。

**Comment by LoveU3tHousand2** [comment]
是的！ 所以说在现有的模型上没有任何办法了是吗， 按您的估计 这个gpu要提升到什么程度才能正常处理最大3500x2000左右的图片啊， 目前的限制是只要超过了h >700 or w>700 就会错误  目前是一张3070显卡 8g。        32g吗

**Comment by LoveU3tHousand2** [comment]
大佬， 我又来了，  现在基本上我的抠图 在分割部分 已经达到要求了， 目前的问题是m_net感觉效果不佳，  目前有一个疑问 就是L_p = L_alpha + L_compositional 中的L_compositional ==｜｜Cg - Cp｜｜ 我对这个Cg到底是什么有些疑惑。  这个仓库的L_compositional中和DIM中的计算方式不一样， 这个作者是用torch.cat((alpha_gt, alpha_gt, alpha_gt) , 1) * img. 去计算Cg t的值，  而DIM中甚至还要用到bg， 所以我很疑惑 这个仓库的作者的这个L_compositional计算方式对吗？ 因为我在训练m_net的过程中把L_compositional打印出来如下，  就很奇怪。  
![image](https://user-images.githubusercontent.com/69668084/124244841-f4279200-db51-11eb-8411-910029f72b7b.png)

**Comment by BarCodeReader** [comment]
> 大佬， 我又来了，  现在基本上我的抠图 在分割部分 已经达到要求了， 目前的问题是m_net感觉效果不佳，  目前有一个疑问 就是L_p = L_alpha + L_compositional 中的L_compositional ==｜｜Cg - Cp｜｜ 我对这个Cg到底是什么有些疑惑。  这个仓库的L_compositional中和DIM中的计算方式不一样， 这个作者是用torch.cat((alpha_gt, alpha_gt, alpha_gt) , 1) * img. 去计算Cg t的值，  而DIM中甚至还要用到bg， 所以我很疑惑 这个仓库的作者的这个L_compositional计算方式对吗？ 因为我在训练m_net的过程中把L_compositional打印出来如下，  就很奇怪。  
> 
> ![image](https://user-images.githubusercontent.com/69668084/124244841-f4279200-db51-11eb-8411-910029f72b7b.png)
> 
> 

最近有点忙，我下个月能重新做这个项目。到时候看能不能有些突破。

你说的这个loss计算看作者具体实现了，如果仅仅比较alpha和前景，相乘是可以的。DIM是算上背景然后考虑整体协调度来进行loss的计算。看你需要哪一个而已。

**Comment by lucky-zjj** [comment]
> 这样的TNet？这样的ANet? ![image](https://user-images.githubusercontent.com/39732241/111504493-ad6c8280-8782-11eb-9408-c1ed94e80686.png)
> 
> ![image](https://user-images.githubusercontent.com/39732241/111328343-1e3f6c00-86a9-11eb-8a36-28b74751f7c5.png) 别想着白嫖了， 自己动手吧。 这个repo的作者已经给你们平了90%的路了，兄弟们。

没显卡呀，呜呜
